Stack/Opc.Ua.Core/Stack/Tcp/TcpTransportListener.cs

```diff
@@ -364,6 +378,25 @@ public void Start()
             }
         }
 
+        /// <summary>
+        /// The callback timer which detects stale channels
+        /// </summary>
+        /// <param name="state"></param>
+        private void DetectInactiveChannels(object state = null)
+        {
+            int tickCount = HiResClock.TickCount;
+            lock(m_lock)
+            {
+                foreach (var chEntry in m_channels)
+                {
+                    if (tickCount - chEntry.Value.LastActiveTime > m_quotas.ChannelLifetime)
```

Comment:
I thought here only `Inactive` or `Opening` channels would time out.

Reply:
Here, instead of using a general lock object I would recommend locking only the m_channels using a read/write lock.